mani_skill/envs/tasks/book_to_shelf.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+from collections import OrderedDict
+from typing import Dict, Union
+
+import numpy as np
+import torch
+import torch.random
+from transforms3d.euler import euler2quat
+import sapien
+import sapien.physx as physx
+import sapien.render
+import transforms3d
+
+from mani_skill.agents.robots import Fetch, Panda, Xmate3Robotiq
+from mani_skill.envs.sapien_env import BaseEnv
+from mani_skill.sensors.camera import CameraConfig
+from mani_skill.utils.building import actors
+from mani_skill.utils.geometry import rotation_conversions
+from mani_skill.utils.registration import register_env
+from mani_skill.utils.sapien_utils import look_at
+from mani_skill.utils.scene_builder.table.table_scene_builder import TableSceneBuilder
+from mani_skill.utils.structs.pose import Pose
+from mani_skill.utils.structs.types import Array
+
+
+@register_env("BookToShelf-v0", max_episode_steps=50)
+class BookToShelfEnv(BaseEnv):
+    SUPPORTED_REWARD_MODES = [
+        "sparse",
+        "none",
+    ]  # TODO add a denser reward for this later
+    SUPPORTED_ROBOTS = ["panda", "xmate3_robotiq", "fetch"]
+    agent: Union[Panda, Xmate3Robotiq, Fetch]
+
+    def __init__(self, *args, robot_uids="panda", robot_init_qpos_noise=0.02, **kwargs):
+        self.robot_init_qpos_noise = robot_init_qpos_noise
+        super().__init__(*args, robot_uids=robot_uids, **kwargs)
+
+    @property
+    def _sensor_configs(self):
+        pose = look_at(eye=[0.3, 0, 0.6], target=[-0.1, 0, 0.1])
+        return [CameraConfig("base_camera", pose, 128, 128, np.pi / 2, 0.01, 100)]
+
+    @property
+    def _human_render_camera_configs(self):
+        pose = look_at([0.6, 0.7, 0.6], [0.0, 0.0, 0.35])
+        return CameraConfig("render_camera", pose, 512, 512, 1, 0.01, 100)
+
+    def _load_scene(self):
+        self.table_scene = TableSceneBuilder(
+            env=self, robot_init_qpos_noise=self.robot_init_qpos_noise
+        )
+        self.table_scene.build()
+
+        self.book = actors.build_actor_ai2_helper(
+            model_id="Book_14",
```

Comment:
I think the task idea is nice but looks a little too strange (the shelf is longer than the table and stretches off of it).

Could we somehow make a scene where there is a tall bookshelf situated right next to the table? E.g. like a shelf against a wall and right in front of the shelf is the table and the robot is there.